common/lib/types/protocolmessage.js

```diff
@@ -71,6 +73,30 @@ var ProtocolMessage = (function() {
 		return this.flags && (this.flags & flags.MODE_ALL);
 	};
 
+	ProtocolMessage.prototype.encodeModesToFlags = function(modes) {
+		var self = this;
+		Utils.arrForEach(modes, function(mode) {
+			self.setFlag(mode);
+		});
+	};
+
+	ProtocolMessage.prototype.decodeModesFromFlags = function() {
+		var modes = [],
+			self = this;
+		Utils.arrForEach([
```

Comment:
This array of modes is duplicated.  Might be worth keeping in one place

Reply:
Fixed in https://github.com/ably/ably-js/pull/613/commits/b727b65b35ef4d05a66bc1902d886d1ab1286ce6